lesson_19_2/test_app.py

```diff
@@ -0,0 +1,36 @@
+import os
+import requests
+import tempfile
+
+def test_upload_retrieve_image():
+    # Create a temporary directory
+    with tempfile.TemporaryDirectory() as temp_dir:
+        # Create a temporary image file
+        temp_file_path = os.path.join(temp_dir, 'image.jpg')
+        with open(temp_file_path, 'wb') as temp_file:
+            # Upload the image
+            response = requests.post('http://127.0.0.1:8080/upload', files={'image': open(temp_file_path, 'rb')})
+            assert response.status_code == 201
```

Comment:
It is better to use 3rd party libraries for assertions. For ex. assertpy. Also it is always better to add failed state descriptions  for your test cases assertions.